src/Select/Select.js

```diff
@@ -26,18 +28,124 @@ const Select = React.forwardRef(({
 
     const handleClick = (e) => {
         setIsExpanded(!isExpanded);
         onClick(e);
     };
 
+    const handleSelect = (e, option) => {
+        setIsExpanded(false);
+        setSelectedOptionKey(option.key);
+        onSelect(e, option);
+    };
+
+    const isKeyboardEvent = (e) => {
+        // we are able to use e.details for modern browsers, but IE11 doesn't work the same way (big surprise)
+        // for IE11, we can make use of e.screenX and e.screenY
+        // the screenX and screenY properties can't be used as a global replacement
+        // because Safari doesn't work the same way for those (of course)
+        const browser = detect();
+
+        if (browser && browser.name === 'ie') {
+            return !e.screenX && !e.screenY;
+        }
+
+        return e.detail === 0;
+    };
+
+
+    const getFocusableMenuItems = () => [...ulRef.current.children];
+
+    const getItemsAndActiveIndex = () => {
+        const items = getFocusableMenuItems();
+        const activeIndex = items.indexOf(document.activeElement);
+
+        return { items, activeIndex };
+    };
+
+    const focusFirst = () => {
+        const { items } = getItemsAndActiveIndex();
+        if (items.length === 0) {
+            return;
+        }
+        if (isKeyboardEvent) {
+            let selectedItem = items.filter(listItem => listItem.className === 'fd-list__item is-selected');
+            selectedItem.length ? setFocusedElement(tryFocus(selectedItem[0])) :
+                setFocusedElement(tryFocus(items[0]));
+        }
+    };
+
+    useEffect(() => {
+        if (isExpanded) focusFirst();
+    }, [isExpanded]);
+
+    useEffect(() => {
+        focusedElement && focusedElement.focus();
+    }, [focusedElement]);
+
+    const focusNext = () => {
+        const { items, activeIndex } = getItemsAndActiveIndex();
+        if (items.length === 0) return;
+
+        if (activeIndex !== items.length - 1) {
+            setFocusedElement(tryFocus(items[activeIndex + 1]));
+        } else {
+            setFocusedElement(tryFocus(items[0]));
+        }
+    };
+
+    const focusPrevious = () => {
+        const { items, activeIndex } = getItemsAndActiveIndex();
+        if (items.length === 0) return;
+
+        if (activeIndex <= 0) {
+            setFocusedElement(tryFocus(items[items.length - 1]));
+        } else {
+            setFocusedElement(tryFocus(items[activeIndex - 1]));
+        }
+    };
+
+    const handleKeyDown = (e, option) => {
+        switch (keycode(e)) {
+            case 'esc':
+            case 'tab':
+                e.stopPropagation();
+                setIsExpanded(false);
+                setFocusedElement(tryFocus(divRef.current));
+                break;
+            case 'enter':
+            case 'space':
+                e.stopPropagation();
+                handleSelect(e, option);
+                setFocusedElement(tryFocus(divRef.current));
```

Comment:
It looks like the intent of the `enter` and `space` is to close the select.  This isn't happening.
For some reason it is calling the `handleClick` immediately after the `handleSelect` and the `handleClick` is toggling the `isExpanded` state back to true

Reply:
It works as expected if you call `e.preventDefualt();` though